docs/architecture/mlcs.md

```diff
@@ -1,2 +1,58 @@
 (mlcs)=
-# Multi-language components
+# Multi-language components (MLCs)
+
+Authors of Pulumi programs can use [component
+resources](https://www.pulumi.com/docs/concepts/resources/components/) to
+logically group related resources together. For instance, a TypeScript program
+might specify a component that combines AWS and PostgreSQL providers to abstract
+the management of an RDS database and logical databases within it:
+
+```typescript
+import * as aws from "@pulumi/aws"
+import * as postgresql from "@pulumi/postgresql"
+
+class Database extends pulumi.ComponentResource {
+  constructor(name: string, args: DatabaseArgs, opts?: pulumi.ComponentResourceOptions) {
+    super("my:database:Database", name, args, opts)
+
+    const rds = new aws.rds.Instance("my-rds", { ... }, { parent: this })
+    const pg = new postgresql.Database("my-db", { ... }, { parent: this })
+
+    ...
+  }
+}
+```
+
+This component can then be used just like any other Pulumi resource:
+
+```typescript
+const db = new Database("my-db", { ... })
+```
+
+...if the program is written in the same language as the component (in this
+case, TypeScript). This is a shame, since components provide a natural means to
+abstract and reuse infrastructure. Moreover, since under the hood components are
+just managing resources like any other Pulumi program, there is no reason their
+use should be limited to a single language.
```

Comment:
```suggestion
case, TypeScript). In some cases however it would be great if components could 
be reused in multiple languages as they provide a natural means to
abstract and reuse infrastructure.  Moreover, since under the hood components are
just managing resources like any other Pulumi program, so it is possible for them to
be abstracted to more languages.
```

I think components still have their place, so we should probably tone down the language here a bit.  MLCs are great, but there is additional complexity in them, so I'd be reluctant to document components as inferior.

Reply:
Hm. This was not my intention -- good shout. Will change the wording!